File: server/src/routes/workPackages.ts
import { Router } from 'express';
import { getCollection } from '../openproject/client.js';
import { mapWorkPackage } from '../openproject/mappers.js';
import { asyncHandler } from '../middleware/error.js';
import type { Task } from '../openproject/mappers.js';

const router = Router();

router.get(
  '/',
  asyncHandler(async (req, res) => {
    const projectId = req.query.projectId as string | undefined;

    const path = projectId
      ? `/api/v3/projects/${projectId}/work_packages`
      : '/api/v3/work_packages';

    const raw = await getCollection<Record<string, any>>(path);
    const tasks: Task[] = raw.map(mapWorkPackage);

    res.json(tasks);
  }),
);

export default router;

File: server_py/app/routes/work_packages.py
from fastapi import APIRouter

from app.openproject.client import get_collection
from app.openproject.mappers import map_work_package
from app.openproject.schemas import Task


router = APIRouter(prefix="/api/work-packages")


@router.get("")
async def list_work_packages(projectId: str | None = None) -> list[Task]:
    path = f"/api/v3/projects/{projectId}/work_packages" if projectId else "/api/v3/work_packages"
    raw = await get_collection(path)
    return [map_work_package(item) for item in raw]
